scipy/spatial/tests/test_distance.py

```diff
@@ -982,12 +1366,18 @@ def _check_calling_conventions(self, X, metric, eps=1e-07, **kwargs):
             if verbose > 2:
                 print(e_cls.__name__)
                 print(e)
-            assert_raises(e_cls, pdist, X, metric=metric, **kwargs)
-            assert_raises(e_cls, pdist, X, metric=eval(metric), **kwargs)
-            assert_raises(e_cls, pdist, X, metric="test_" + metric, **kwargs)
+            try:
+                assert_raises(Exception, pdist, X, metric=metric, **kwargs)
+                assert_raises(Exception, pdist, X, metric=eval(metric), **kwargs)
+                assert_raises(Exception, pdist, X, metric="test_" + metric, **kwargs)
+            except Exception as e2:
```

Comment:
Why are you simply checking for a generic Exception instead of the same one? They may fail for different reasons and I think that we should be able to check that.